src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -176,8 +176,8 @@ protected boolean selfUpdate() {
     private File installOrUpdateScanner(String dirPath) throws IOException, IntegrationException {
         final File installDirectory = new File(dirPath);
         
-        if (!installDirectory.exists()) {
-            installDirectory.mkdir();
+        if (!installDirectory.exists() && !installDirectory.mkdir()) {
```

Comment:
I would go one step further:
```
        if (!installDirectory.exists() && !installDirectory.mkdir()) {
            return null;
        } else if (!checkInstallationDir(installDirectory.toPath())) {
            return null;
        }
```
to
```
        if (!installDirectory.exists() && !installDirectory.mkdir() && !checkInstallationDir(installDirectory.toPath())) {
            return null;
        }
```

Reply:
Please disregard my above comment and merge as-is. This change came from the community and is valid. Any further improvement should be done in an appropriate PR.

Reply:
Hi @gopannabd I'll open a ticket with the above suggestion. Thank you.